### ISSUE TITLE
Wrong measure text field width in BitmapFontTextRenderer

### ISSUE THREAD

**Issue by Yaniko** [BowlerHatLLC/feathers#1676]
There is an error in measuring white space after a symbol. Current code is like (previousCharData.xAdvance - previousCharData.width) * scale;
Lines
https://github.com/BowlerHatLLC/feathers/blob/master/source/feathers/controls/text/BitmapFontTextRenderer.as#L806
https://github.com/BowlerHatLLC/feathers/blob/master/source/feathers/controls/text/BitmapFontTextRenderer.as#L833
and others.
![metrics](https://user-images.githubusercontent.com/27950327/34457305-ab5f169a-edbd-11e7-8e35-308203c145a1.png)

The right formulae is xAdvance - width - xOffset, because char.width doesn't contain xOffset. 
xAdvance = xOffset + width + someSpaceAfter
Because of that issue some chars (like colon) get cropped from the right in a textfield with TextAlign.RIGHT

**Comment by joshtynjala** [comment]
Thanks, I'll take a look soon!

**Comment by joshtynjala** [comment]
Switching to `xAdvance - width - xOffset` doesn't seem to work in some cases. According to another user who has tried the new code, this change causes BitmapFontTextRenderer to begin truncating text where it hadn't previously. Since that's not good at all, I have reverted this commit for now.

Unfortunately, the bitmap fonts that I've tested with cannot reproduce your issue, and I could not reproduce the new issue either. I guess it depends on the exact bitmap font that you use.

**Comment by Yaniko** [comment]
We use fonts with shadows generated in literra.
Extra truncating means text size become wider. And that means xOffset is negative? Really strange.

**Comment by Yaniko** [comment]
Sorry, I was wrong in my last message. 
Of course strings become wider, because we subtract some positive value from whitespace. It makes sense.
Have you tried textalign.Right?

**Comment by joshtynjala** [comment]
No, I have not been able to reproduce with text aligned to the right either.

**Comment by MaximVaulin** [comment]
A few days ago i tried textalign.Right on label with background and watched how text borders are out of labels background

**Comment by divlaine** [comment]
a few days ago i tried textalign.Right on a label with backgroundskin and saw the text was out from label's borders
